app/controllers/case_contacts/form_controller.rb

```diff
@@ -52,6 +44,13 @@ def set_case_contact
     @case_contact = CaseContact.find(params[:case_contact_id])
   end
 
+  def manage_form_step(step_navigation: false)
+    authorize @case_contact
+    @page = wizard_steps.index(step) + 1
+    @total_pages = steps.count
```

Comment:
This information is necessary for #update if you want to continue seeing the progress bar and navigation buttons after a form/validation error

Reply:
This doesn't depend on any specific info, does it make more sense in a before action?